bundles/CoreBundle/EventListener/ResponseHeaderListener.php

```diff
@@ -57,4 +60,27 @@ public function onKernelResponse(ResponseEvent $event)
             $response->headers->set($header->getKey(), $header->getValues(), $header->getReplace());
         }
     }
+
+    public function onKernelControllerArguments(ControllerArgumentsEvent $event): void
+    {
+        $request = $event->getRequest();
+        /** @var $configurations ResponseHeader */
+        if (!$attributes = $request->attributes->get('_response_header')) {
+            return;
+        }
+
+        foreach ($attributes as $attribute) {
+            if (!$attributes instanceof ResponseHeader) {
+                trigger_deprecation(
+                    'pimcore/pimcore',
+                    '10.6',
+                    'Usage of @ResponseHeader annotation is deprecated. please use #[ResponseHeader] attribute instead.'
+                );
+            }
+        }
+
+        $responseHeaders = array_merge($this->responseHeaderResolver->getResponseHeaders($request), $attributes);
```

Comment:
there's something wrong with this line, the same responseHeader appearing twice.

Reply:
Response header key is repeated for multiple values so check if this is the case.

Reply:
- The [$this->responseHeaderResolver->getResponseHeaders($request)](https://github.com/pimcore/pimcore/blob/a661f034c6bc6b6f4fb06804d1e42e3df8e34db3/lib/Http/Request/Resolver/ResponseHeaderResolver.php#L44) returns `$request->attributes->get('_response_header')` which is the same values as `$attributes` (on line 68). 
- `if (!$attribute instanceof ResponseHeader) {` , still doesn't differenciate the annotation and the attribute, so the deprecation likely never triggers or triggers on wrong circumstances
![image](https://user-images.githubusercontent.com/6014195/216431854-d89fd496-d89b-47f2-a535-93dc8f627626.png)

Reply:
fixed

Reply:
Separated attribute from configuration, now we can trigger deprecation, if deprecated configuration is used.